### PR TITLE
Don't send emails to hubspot that are empty strings

### PR DESCRIPTION
**Pull request by TylerSheffels**
@dannyroberts 

cc: @benrudolph

### CODE REVIEW

**Comment by czue** [comment]
:+1: when tests pass

File: corehq/apps/analytics/tasks.py [hunk 1]
```diff
@@ -231,6 +231,8 @@ def track_periodic_data():
     submit = []
     for user in users_to_domains:
         email = user['email']
+        if not email:
+            continue
```
Comment (dannyroberts) [inline corehq/apps/analytics/tasks.py]: Is this the fix we talked about? Is the fixing of `get_email`'s logic in a different PR?

Comment (TylerSheffels) [inline corehq/apps/analytics/tasks.py]: The `get_email` call isn't actually used in this code path.  The empty emails are being returned from an Elastic query, and there only 8 out of about 4000 that were empty.

Comment (dannyroberts) [inline corehq/apps/analytics/tasks.py]: gotcha

**Pull request merged**